base/collection_n1ql_common.go

```diff
@@ -218,33 +217,35 @@ func BuildDeferredIndexes(s N1QLStore, indexSet []string) error {
 	}
 
 	InfofCtx(context.TODO(), KeyQuery, "Building deferred indexes: %v", deferredIndexes)
-	buildErr := buildIndexes(s, deferredIndexes)
+	buildErr := buildIndexes(bucket, s, deferredIndexes)
 	return buildErr
 }
 
 // BuildIndexes executes a BUILD INDEX statement in the current bucket, using the form:
 //
 //	BUILD INDEX ON `bucket.Name`(`index1`, `index2`, ...)
-func buildIndexes(s N1QLStore, indexNames []string) error {
+func buildIndexes(bucket Bucket, s N1QLStore, indexNames []string) error {
```

Comment:
I think we want to keep N1QLStore if possible

Reply:
I ensured I left the n1ql store in the function parameters so its still used. We need the bucket to be able to call the new WatchForIndexesOnline code.

Reply:
Can we add WatchForIndexes to N1QLStore code?

Reply:
Good point 🤦🏼‍♂️ that should work